pkg/api/v1/defaults.go

```diff
@@ -371,3 +371,15 @@ func SetDefaults_ScaleIOVolumeSource(obj *ScaleIOVolumeSource) {
 		obj.FSType = "xfs"
 	}
 }
+
+func SetDefaults_HugePagesVolumeSource(obj *HugePagesVolumeSource) {
+	if obj.PageSize == "" {
+		obj.PageSize = "2M"
+	}
+	if obj.MaxSize == "" {
```

Comment:
What I was saying wasn't if not to specify size is valid. I was saying is that if users specify as below:
```
    hugePages:
       pageSize: "2M"
       maxSize: ""
       minSize: "100M"
```

This will default to:
```
    hugePages:
       pageSize: "2M"
       maxSize: "2M"
       minSize: "100M"

```
Then the maxSize < minSize, which is invalid.